sympy/solvers/solveset.py

```diff
@@ -548,6 +528,9 @@ def _solveset(f, symbol, domain, _check=False):
         a, h = f.as_independent(symbol)
         m, h = h.as_independent(symbol, as_Add=False)
         f = a/m + h  # XXX condition `m != 0` should be added to soln
+    numer, denom = f.as_numer_denom()
```

Comment:
Thanks @smichr .
I tried few cases present in the `test_solvers.py`. One of the good problem I faced is 
```
In [1]: z = sqrt(2*x + 1)/sqrt(x) - sqrt(2 + 1/x)

In [2]: solve(z,x)
Out[2]: []
``` 
But using this(above changes) method it can't be solved. 
I was trying to simplify this expression before going to `_solveset` but `powsimp`, `ratsimp` and other didn't help.
If you have any ideas then pleased suggest.